pkg/ddl/ingest/backend.go

```diff
@@ -193,28 +188,28 @@ func (bc *litBackendCtx) Flush(indexID int64, mode FlushMode) (flushed, imported
 		return true, false, nil
 	}
 
-	// Use distributed lock if run in distributed mode).
-	if bc.etcdClient != nil {
-		distLockKey := fmt.Sprintf("/tidb/distributeLock/%d/%d", bc.jobID, indexID)
-		se, _ := concurrency.NewSession(bc.etcdClient)
-		mu, err := acquireLock(bc.ctx, se, distLockKey)
+	if bc.useDistributedLock {
+		store := newDistLockDataStore(bc.store)
+		dKey := strconv.Itoa(int(bc.jobID))
+		dLock := meta.NewDistributedLockBuilder().
+			SetBackoff(5*time.Second).
+			SetLease(2*time.Minute).
+			Build(bc.ctx, store, bc.instanceID, dKey)
+		err = dLock.Lock()
 		if err != nil {
 			return true, false, err
 		}
 		logutil.Logger(bc.ctx).Info("acquire distributed flush lock success", zap.Int64("jobID", bc.jobID))
 		defer func() {
-			err = mu.Unlock(bc.ctx)
+			err := dLock.Unlock()
 			if err != nil {
 				logutil.Logger(bc.ctx).Warn("release distributed flush lock error", zap.Error(err), zap.Int64("jobID", bc.jobID))
 			} else {
 				logutil.Logger(bc.ctx).Info("release distributed flush lock success", zap.Int64("jobID", bc.jobID))
 			}
-			err = se.Close()
-			if err != nil {
-				logutil.Logger(bc.ctx).Warn("close session error", zap.Error(err))
-			}
 		}()
 	}
+
 	err = bc.unsafeImportAndReset(ei)
```

Comment:
I don't understand how distributed lock solves the problem. If node A acquired the lock, started an import, then lock is expired and node B now acquired the lock. A has no way to cancel its running import task after B starts? If we tolerate two nodes run import task simultaneously for a short time, we can use more simple way to implement it?

Reply:
Could you explain the simple way in details?

> cancel the task when the task owner node is not itself

Then we need a way to persist the state of "task owner node". This PR stores this state to a meta key.